packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -786,20 +741,22 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
         // could have got nacked.
         if (context.proposalHandle !== this.blobsShaProposalHandle) {
```

Comment:
Do we ever hit this branch? It feels like it's a protection branch, but client should never get here.
If so, can we add an error logging statement?

Reply:
No, we can get here when the last uploaded summary gets nacked. In that case, we would have updated our blobShaProposedHandle but context.proposedHandle would remain same when the summary upload is done again(due to nack). So in that case we would hit this branch. 
But now unlike clearing the cache, I have made another backupCache and we would just reinitialize the mainCache with backup cache so that we would still be able to do the deduping.

Reply:
But i can still add a telemetry event(not  the error one).